build_tools/cmake/build_host_tools.sh

```diff
@@ -45,6 +45,9 @@ declare -a CMAKE_ARGS=(
   "-DIREE_BUILD_TESTS=OFF"
   "-DIREE_BUILD_SAMPLES=OFF"
 
+  # Enable building Tracy capture and profiler tools.
+  "-DIREE_BUILD_TRACY=ON"
```

Comment:
I also see

```
[cmake] -- Could not find Tracy dependencies (Tracy server will not be built).
[cmake] -- To build Tracy, install packages libzstd, and tbb (on Ubuntu/Debian, 'apt install libcapstone-dev libtbb-dev libzstd-dev')
[cmake] CMake Error at CMakeLists.txt:780 (message):
[cmake]   Could not build Tracy.  Either unset IREE_BUILD_TRACY or look for missing
[cmake]   dependencies above and install them.
```

on my dev machine, and I'd like developers to be able to run this script without installing all that, since I reference it in some docs: https://github.com/iree-org/iree/blob/2118e024e24ea42dbc45bb50891b0462ae5dd640/experimental/web/sample_static/build_sample.sh#L10-L13

That being said,
* We should be testing this build on presubmit, since our nightly release build uses it and we want early failure notifications
* I do this this script is the right place to test this, choosing between other scripts used in CI pipelines

So maybe we could make it fail gracefully here? Or just accept that our presubmit coverage will be sparse since this isn't a critical component?

Reply:
Generally I think enhancing pre-submit testing to catch issues earlier is good, if the testing isn't time consuming. But here Tracy is essentially a third-party tool that aren't always needed like the other host tools and it carries quite a lot of dependencies to probe. I think given the importance and complexity it's fine to omit it to avoid bringing in lots of logic in the `build_host_tools.sh` script, particularly given that it's meant for general user usage too.

I think another way is to turn `IREE_BUILD_TRACY` as a "soft" control; if we find missing dependencies then just force it to `OFF`. That way we don't need to duplicate all library probe logic in the bash script again.

Reply:
How about we make `build_host_tools.sh` optionally set `-DIREE_BUILD_TRACY=ON` based on an environment variable, similar to how we have test scripts optionally run Vulkan/CUDA/LLVM CPU tests?

Reply:
Eh, Tracy has so many recent dependencies (notably glfw and zstd) that I don't see this working out until we update the Docker images used to build this to a more recent OS version.

In principle we could add an option like https://github.com/iree-org/iree/blob/6b8387ca5c52a477ae6cad500749d8efe008264b/build_tools/cmake/build_host_tools.sh#L21 for `IREE_BUILD_TRACY`, but we'd need to use the release Docker image or a new Docker image (not one of those derived from `base`)

Reply:
For posterity, here's what I tried: https://github.com/ScottTodd/iree/commit/d1f4695d0e1a6e3ff16e2633393bf77396985c43